src/test/java/com/icthh/xm/ms/balance/service/BalanceServiceUnitTest.java

```diff
@@ -1825,4 +1832,395 @@ public void setIsLastBalanceChangeEvent() {
         assertFalse(balanceEventForUpdate.getLast());
     }
 
+    @Test(expected = EntityNotFoundException.class)
+    public void shouldThrowExceptionWhenBalanceNotFound() {
+        when(balanceRepository.findOneByIdForUpdate(5L)).thenReturn(Optional.empty());
+        balanceService.revertBalanceOperation(new RevertBalanceOperationRequest().setBalanceId(5L));
+        verifyNoMoreInteractions(balanceChangeEventRepository);
+    }
+
+    @Test
+    public void shouldSuccessWhenReturnDuplicateOperation() {
+        long balanceId = 123456789;
+        createBalance(balanceId);
+
+        RevertBalanceOperationRequest revertBalanceOperationRequest = new RevertBalanceOperationRequest();
+        String duplicateUuid = UUID.randomUUID().toString();
+        revertBalanceOperationRequest.setUuid(duplicateUuid);
+        revertBalanceOperationRequest.setBalanceId(balanceId);
+
+        BalanceChangeEvent existBalanceChangeEvent = createBalanceEvent("90", 1L, RELOAD, "100", "10");
+        existBalanceChangeEvent.setOperationId(duplicateUuid);
+        when(balanceChangeEventRepository.findBalanceChangeEventsByOperationId(duplicateUuid))
+            .thenReturn(List.of(existBalanceChangeEvent));
+
+        BalanceChangeEventDto expectedEvent = balanceChangeEventMapper.toDto(existBalanceChangeEvent);
+        BalanceChangeEventDto actualEvent = balanceService.revertBalanceOperation(revertBalanceOperationRequest);
+
+        assertEquals(expectedEvent, actualEvent);
+    }
+
+    @Test(expected = EntityNotFoundException.class)
+    public void shouldThrowExceptionWhenEventToRevertNotFound() {
+        long balanceId = 123456789;
+        String currentUuid = UUID.randomUUID().toString();
+        String uuidToRevert = UUID.randomUUID().toString();
+
+        createBalance(balanceId);
+
+        RevertBalanceOperationRequest revertBalanceOperationRequest = new RevertBalanceOperationRequest();
+        revertBalanceOperationRequest.setBalanceId(balanceId);
+        revertBalanceOperationRequest.setUuid(currentUuid);
+        revertBalanceOperationRequest.setUuidToRevert(uuidToRevert);
+
+        when(balanceChangeEventRepository.findBalanceChangeEventsByOperationId(currentUuid))
+            .thenReturn(emptyList());
+        when(balanceChangeEventRepository.findBalanceChangeEventsByOperationId(uuidToRevert))
```

Comment:
pls check that no unexpected method invokations
and in next tests too

Reply:
Fixed